connection.go

```diff
@@ -196,6 +196,13 @@ func (c *Connection) handleError(err error) {
 		return
 	}
 
+	c.mutex.Lock()
+	if c.closing {
+		c.mutex.Unlock()
+		return
+	}
+	c.mutex.Unlock()
```

Comment:
Is it easier to write the code as this? 

```suggestion
	c.mutex.Lock()
	closed := c.closing
	c.mutex.Unlock()
	if closed {		
		return
	}
```